core/src/main/java/org/jboss/pnc/build/finder/protobuf/KojiArchiveInfoAdapter.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (C) 2017 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.pnc.build.finder.protobuf;
+
+import org.infinispan.protostream.annotations.ProtoAdapter;
+import org.infinispan.protostream.annotations.ProtoFactory;
+import org.infinispan.protostream.annotations.ProtoField;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.redhat.red.build.koji.model.json.util.KojiObjectMapper;
+import com.redhat.red.build.koji.model.xmlrpc.KojiArchiveInfo;
+
+/**
+ * Protostream adapter to be able to properly marshall / unmarshall KojiArchiveInfo.
+ *
+ * This class is defined in Kojiji and can't be easily modified to add Proto annotations. Instead, the marshalling
+ * process involves converting KojiArchiveInfo object into JSON and stored in Protobuf as a string. The unmarshalling
+ * process involves reading the JSON string from Protobuf and converting it back to a KojiArchiveInfo object.
+ */
+@ProtoAdapter(KojiArchiveInfo.class)
+public class KojiArchiveInfoAdapter {
+    private static final ObjectMapper OBJECT_MAPPER = new KojiObjectMapper();
```

Comment:
We also have `BuildFinderObjectMapper` which extends `KojiObjectMapper` and you should probably use that.

Though not your fault and it should perhaps be in a separate PR, I was hoping that we had a single shared instance somewhere of the mapper, but we do not. Perhaps it belongs in `KojiJSONUtils`. We never change it, but with a shared instance, I don't think there's away to ensure that it is read-only. Thoughts?

Reply:
Sounds good to me

Reply:
I am reverting that change to share object mapper. For example, `BuildConfig.java` adds additional configuration in the static initializer that breaks 'KojiArchiveInfoAdapter' as a result.

```[WARN] Built archive xxx
java.lang.ClassCastException: class java.util.LinkedHashMap cannot be cast to class java.lang.String (java.util.LinkedHashMap and java.lang.String are in module java.base of loader 'bootstrap')
	at com.redhat.red.build.koji.model.xmlrpc.KojiMavenBuildRequest.lambda$new$2(KojiMavenBuildRequest.java:72)
	at java.base/java.util.stream.Collectors.lambda$uniqKeysMapAccumulator$1(Collectors.java:178)
	at java.base/java.util.stream.ReduceOps$3ReducingSink.accept(ReduceOps.java:169)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
	at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at com.redhat.red.build.koji.model.xmlrpc.KojiMavenBuildRequest.<init>(KojiMavenBuildRequest.java:71)
	at com.redhat.red.build.koji.model.xmlrpc.KojiTaskRequest.asMavenBuildRequest(KojiTaskRequest.java:60)
	at org.jboss.pnc.build.finder.report.HTMLReport.lambda$toHTML$5(HTMLReport.java:330)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at j2html.tags.ContainerTag.with(ContainerTag.java:106)
	at j2html.TagCreator.each(TagCreator.java:118)
	at org.jboss.pnc.build.finder.report.HTMLReport.toHTML(HTMLReport.java:263)
	at org.jboss.pnc.build.finder.report.HTMLReport.renderText(HTMLReport.java:379)
	at org.jboss.pnc.build.finder.report.Report.outputHTML(Report.java:87)
	at org.jboss.pnc.build.finder.report.Report.generateReports(Report.java:65)
	at org.jboss.pnc.build.finder.cli.Main.call(Main.java:818)
	at org.jboss.pnc.build.finder.cli.Main.call(Main.java:93)
	at picocli.CommandLine.executeUserObject(CommandLine.java:1953)
	at picocli.CommandLine.access$1300(CommandLine.java:145)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2352)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2346)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2311)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2179)
	at picocli.CommandLine.execute(CommandLine.java:2078)
	at org.jboss.pnc.build.finder.cli.Main.main(Main.java:234)


```

Reply:
Yes, OK. Maybe it would be possible to use it after all common configuration is merged in a separate PR. Yes, if it's public someone can also accidentally break it for everyone, perhaps it can be made package-private still.